samples/QuickstartWeatherServer/Tools/WeatherTools.cs

```diff
@@ -0,0 +1,84 @@
+﻿using ModelContextProtocol.Server;
+using System.ComponentModel;
+using System.Net.Http.Headers;
+using System.Text.Json;
+
+namespace QuickstartWeatherServer.Tools;
+
+[McpServerToolType]
+public static class WeatherTools
+{
+    [McpServerTool, Description("Get weather alerts for a US state.")]
+    public static async Task<string> GetAlerts(
+        [Description("The US state to get alerts for.")] string state)
+    {
+        using HttpClient client = GetWeatherClient();

```

Comment:
To avoid having a getting-started sample that shows the antipattern of creating/destroying an HttpClient (and its connection pool) on every use, could this instead either a) accept the HttpClient as a parameter to GetAlerts such that it can be injected via DI, or b) have the HttpClient cached in a static field?

Reply:
Added via DI - there's already a services provider so it may as well get used.